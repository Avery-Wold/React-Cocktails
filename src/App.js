import React, { Component } from 'react';
import './App.scss';
import Main from './Components/MainPage/Main';

export class App extends Component {
  render() {
    return (
      <Main />
    )
  }
}

export default App
