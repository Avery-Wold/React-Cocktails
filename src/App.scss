@import './Styles/variables.scss';

.App {
  text-align: center;
  font-family: Arial, Helvetica, sans-serif;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
  pointer-events: none;
}

.App-header {
  background-color: $primary-color;
  height: 200px;
  display: flex;
  flex-direction: column;
  font-size: calc(10px + 2vmin);
  color: white;
}

.pull-right {
  margin-right: 20%;
}

.title {
  font-size: calc(20px + 2vmin);
  justify-content: center;

}

.App-link {
  color:  $secondary-color;
}

.App-intro {
  margin-top: 20px;
  margin-left: auto;
  margin-right: auto;
  width: $table-width;
  font-size: 20px;
}

p {
  font-size: 20px;
}

.btn {
  width: 200px;
  height: 50px;
  font-size: 20px;
  border-radius: 5px;
  margin-top: 20px;
  background-color: $button-background !important;
  border: 1px solid #ccc;

  &:hover {
    background: $button-hover !important;
  }
}

.btn-fave {
  width: 200px;
  height: 50px;
  font-size: 20px;
  border-radius: 5px;
  margin-top: 20px;
  background-color: $button-background !important;
  border: 1px solid #ccc;
  position: relative;
  left: -10px;

  &:hover {
    background: $button-hover !important;
  }
}

.select-drink {
  width: 100%;
  float: left;
  box-sizing: inherit;
  position: relative;
  bottom: 190px;
}

.drink-selection {
  float: left;
  margin-left: 20px;
  width: 31%;
}

.drink-img {
  width: $image-width;
  height: $image-height;
  float:left;
  border-radius: 5px;
  margin-left: 30px;
  // margin-top: 10px;
}

.drink-home1 {
  width: $image-width;
  height: $image-height;
  float:left;
  border-radius: 5px;
  margin-left: 30px;
  margin-top: 30px;
}

.drink-home2 {
  width: $image-width;
  height: $image-height;
  float:right;
  border-radius: 5px;
  margin-right: 30px;
  margin-top: 30px;
  position: relative;
  top: -205px;
}

.drink {
  margin-top: 30px;
  margin-left: auto;
  margin-right: auto;
}

.table {
  width: $table-width !important;
  margin-left: auto;
  margin-right: auto;
  font-size: calc(10px + 2vmin);
}

input[type=text] {
  width: 300px;
  padding: 12px 20px;
  margin: 8px 0;
  box-sizing: border-box;
  border-radius: 5px;
}

.NavbarToggler {
  float: right;
}

.drink-list-div {
  padding: 20px;
  height: 10%;
  float: left;
  width: 33%;
  box-sizing: inherit;
}

.container {
  border: 1px solid #ccc;
  box-sizing: inherit;
  padding-bottom: 20px;
}

li {
  padding: 10px;
  margin-right: 10px;
  margin-left: 10px;
  display: inline-block;
  font-size: 20px;
  text-align: center;
}

.drink-list {
  width: $image-width;
  height: $image-height;
  float:left;
  border-radius: 5px;
  margin-top: 10px;
  margin-left: 20px;
}

.table h1 {
  font-size: calc(30px + 2vmin);
}

.form-errors {
  color: $form-error;
}