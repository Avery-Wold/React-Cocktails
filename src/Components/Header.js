import React, { Component } from 'react';
import { Collapse, Navbar, NavbarToggler, NavbarBrand, Nav, NavItem, NavLink } from 'reactstrap';
import 'bootstrap/dist/css/bootstrap.css';
import '../App.scss';

export default class Example extends Component {
  constructor(props) {
    super(props);

    this.state = {
      collapsed: true
    };
  }

  toggleNavbar = () => {
    this.setState(state => ({
      collapsed: !this.state.collapsed
    }));
  }

  render() {
    return (
      <div className="navbar navbar-expand-md bg-dark navbar-dark">
        <Navbar >
          <NavbarBrand href="/" className="title">Making Cocktails With Avery</NavbarBrand>
          <div className="pull-right">
          <NavbarToggler onClick={this.toggleNavbar} className="navbar-collapse"/>
          <Collapse isOpen={!this.state.collapsed} >
            <Nav navbar>
              <NavItem>
                <NavLink href="https://github.com/Avery-Wold/React-Masters-Spring-2019">GitHub</NavLink>
              </NavItem>
            </Nav>
          </Collapse>
          </div>
        </Navbar>
      </div>
    );
  }
}
