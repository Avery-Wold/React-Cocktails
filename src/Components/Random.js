import React, { Component } from 'react';
import '../App.scss';
import { Drink } from './Drink';
import { getRandom } from '../Actions/DrinkInfo';
import { connect } from 'react-redux';

class Random extends Component {
    constructor(props) {
        super(props);
        
        this.GetRandomCocktail = this.GetRandomCocktail.bind(this);
    }

    GetRandomCocktail(e) {
        e.preventDefault();
        const {dispatch} = this.props;
        dispatch((getRandom()));    
    }

    render() {
        return (
            <div className="drink-selection">
                <p>
                Just want a random drink?
                </p>
                <p>
                <button className="btn" onClick={e => this.GetRandomCocktail(e)}>
                    Get Random Drink
                </button>
                </p>
            </div>
        )
    }
}
  
const mapStateToProps = (state) => {
    return {
        drink: state.DrinkInfo.drink,
        glass: state.DrinkInfo.glass,
        instructions: state.DrinkInfo.instructions,
        ingredient1: state.DrinkInfo.ingredient1,
        ingredient2: state.DrinkInfo.ingredient2,
        ingredient3: state.DrinkInfo.ingredient3,
        picture: state.DrinkInfo.picture,
        measure1: state.DrinkInfo.measure1,
        measure2: state.DrinkInfo.measure2,
        measure3: state.DrinkInfo.measure3,
        isFave: state.DrinkInfo.isFave
    }
}

export default connect(mapStateToProps) (Random)
