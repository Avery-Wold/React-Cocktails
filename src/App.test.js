import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Header from './Components/Header';
import { shallow, mount, configure } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import { Main } from './Components/MainPage/Main';
import Random from './Components/Random';

configure({adapter: new Adapter()});

describe("App", () => {
  
  it("should render correctly", () => {
    const shallowApp = shallow(<App />);
    expect(shallowApp).toMatchSnapshot();
  });

  it("main should render correctly", () => {
    const shallowApp = shallow(<Main />);
    expect(shallowApp).toMatchSnapshot();
  });
});