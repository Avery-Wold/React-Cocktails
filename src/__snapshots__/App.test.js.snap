// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App main should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Example />,
        <img
          alt="cocktail"
          className="drink-home1"
          src="https://www.thecocktaildb.com/images/media/drink/mtpxgk1504373297.jpg"
        />,
        <div
          className="App-intro"
        >
          <h1>
            Welcome!
          </h1>
          <p>
            Mixology is popular, have you been wanting to try to mix your own cocktails?  Let's get started!  You will be able to select a spirit you would like to use, or enter the name of a cocktail you like and want to try making yourself.  If you just want to see what Avery's favorite is, just click the button below!
          </p>
        </div>,
        <img
          alt="cocktail"
          className="drink-home2"
          src="https://www.thecocktaildb.com/images/media/drink/vrwquq1478252802.jpg"
        />,
        <button
          className="btn-fave"
          onClick={[Function]}
        >
          Avery's favorite
        </button>,
        <div
          className="select-drink"
        >
          <Connect(CocktailSearch) />
          <Connect(Random) />
          <Connect(SpiritSearch) />
        </div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "cocktail",
          "className": "drink-home1",
          "src": "https://www.thecocktaildb.com/images/media/drink/mtpxgk1504373297.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Welcome!
            </h1>,
            <p>
              Mixology is popular, have you been wanting to try to mix your own cocktails?  Let's get started!  You will be able to select a spirit you would like to use, or enter the name of a cocktail you like and want to try making yourself.  If you just want to see what Avery's favorite is, just click the button below!
            </p>,
          ],
          "className": "App-intro",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome!",
            },
            "ref": null,
            "rendered": "Welcome!",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Mixology is popular, have you been wanting to try to mix your own cocktails?  Let's get started!  You will be able to select a spirit you would like to use, or enter the name of a cocktail you like and want to try making yourself.  If you just want to see what Avery's favorite is, just click the button below!",
            },
            "ref": null,
            "rendered": "Mixology is popular, have you been wanting to try to mix your own cocktails?  Let's get started!  You will be able to select a spirit you would like to use, or enter the name of a cocktail you like and want to try making yourself.  If you just want to see what Avery's favorite is, just click the button below!",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "cocktail",
          "className": "drink-home2",
          "src": "https://www.thecocktaildb.com/images/media/drink/vrwquq1478252802.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Avery's favorite",
          "className": "btn-fave",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Avery's favorite",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(CocktailSearch) />,
            <Connect(Random) />,
            <Connect(SpiritSearch) />,
          ],
          "className": "select-drink",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Example />,
          <img
            alt="cocktail"
            className="drink-home1"
            src="https://www.thecocktaildb.com/images/media/drink/mtpxgk1504373297.jpg"
          />,
          <div
            className="App-intro"
          >
            <h1>
              Welcome!
            </h1>
            <p>
              Mixology is popular, have you been wanting to try to mix your own cocktails?  Let's get started!  You will be able to select a spirit you would like to use, or enter the name of a cocktail you like and want to try making yourself.  If you just want to see what Avery's favorite is, just click the button below!
            </p>
          </div>,
          <img
            alt="cocktail"
            className="drink-home2"
            src="https://www.thecocktaildb.com/images/media/drink/vrwquq1478252802.jpg"
          />,
          <button
            className="btn-fave"
            onClick={[Function]}
          >
            Avery's favorite
          </button>,
          <div
            className="select-drink"
          >
            <Connect(CocktailSearch) />
            <Connect(Random) />
            <Connect(SpiritSearch) />
          </div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "cocktail",
            "className": "drink-home1",
            "src": "https://www.thecocktaildb.com/images/media/drink/mtpxgk1504373297.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Welcome!
              </h1>,
              <p>
                Mixology is popular, have you been wanting to try to mix your own cocktails?  Let's get started!  You will be able to select a spirit you would like to use, or enter the name of a cocktail you like and want to try making yourself.  If you just want to see what Avery's favorite is, just click the button below!
              </p>,
            ],
            "className": "App-intro",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome!",
              },
              "ref": null,
              "rendered": "Welcome!",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Mixology is popular, have you been wanting to try to mix your own cocktails?  Let's get started!  You will be able to select a spirit you would like to use, or enter the name of a cocktail you like and want to try making yourself.  If you just want to see what Avery's favorite is, just click the button below!",
              },
              "ref": null,
              "rendered": "Mixology is popular, have you been wanting to try to mix your own cocktails?  Let's get started!  You will be able to select a spirit you would like to use, or enter the name of a cocktail you like and want to try making yourself.  If you just want to see what Avery's favorite is, just click the button below!",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "cocktail",
            "className": "drink-home2",
            "src": "https://www.thecocktaildb.com/images/media/drink/vrwquq1478252802.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Avery's favorite",
            "className": "btn-fave",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Avery's favorite",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(CocktailSearch) />,
              <Connect(Random) />,
              <Connect(SpiritSearch) />,
            ],
            "className": "select-drink",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`App random should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Random) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`App should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`App should render correctly 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Main) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`App should render correctly 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Random) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
